### DISCUSSION TITLE
Optional Support class give null with nested getters...

### DISCUSSION THREAD

**Discussion by cf-git** [laravel/framework#34516] [General]
When i'm use related models i'd get null with existing result

```php
<?php //test.php
dd(
 $staticPage->page->test, //returns 'Ok!'
 optional($staticPage->page)->test // returns null
);
```

```php
<?php

namespace App\Models;

class StaticPage extends LocalizableModel
{
        /**
     * @return MorphOne
     */
    public function page()
    {
        return $this->morphOne(Page::class, 'pageable');
    }
}
```

```php
<?php

namespace App\Models;

class Page extends LocalizableModel
{
    
}
```

```php
<?php
abstract class LocalizableModel extends Model
{
   /**
     * Magic method for retrieving a missing attribute
     *
     * @param string $attribute
     * @return mixed
     */
    public function __get($attribute)
    {
        if (in_array($attribute, ['test'])) {
           return 'Ok!';
        }

        return parent::__get($attribute);
    }
}
```

```php
<?php
namespace Illuminate\Support;

class Optional implements ArrayAccess
{
    public function __get($key)
    {
        if (is_object($this->value)) {
            // this code do not check result exsists, but it check variable exists
            return $this->value->{$key} ?? null;
        }
    }
}
```

Used:
- Laravel Framework 7.22 
- PHP 7.3.20

**Comment by driesvints** [comment]
Hi there,

Thanks for reporting but it looks like this is a question which can be asked on a support channel. Please only use this issue tracker for reporting bugs with the library itself. 

You can post support questions on our discussions tab instead so I'm moving it over there. I won't be able to help you out personally but others who are watching this repository can help you out.

Alternatively, you can post this question on one of the following channels:

- [Laracasts Forums](https://laracasts.com/discuss)
- [Laravel.io Forums](https://laravel.io/forum)
- [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
- [Discord](https://discordapp.com/invite/KxwQuKb)
- [Larachat](https://larachat.co)
- [IRC](https://webchat.freenode.net/?nick=laravelnewbie&channels=%23laravel&prompt=1)

Thanks.

**Comment by cf-git** [comment]
Thanks, but ... It's not a question. Maybe I choose the wrong words. I'm already solve this trouble for myself. 
I think that many developers expect the behavior to be predictable ... that is, optional attributes which has manual handlers (i mean, can not be defined explicitly)
Models or classes which has getters and setters.
I propose just update handler which do not check existing of value but check result.

I just propose add this in future
Thank, you!
